runner/runner.go

```diff
@@ -464,6 +473,78 @@ func runEvents(workerID int, events <-chan event) {
 	gEventWorkersRunning[workerID-1] = false
 }
 
+func runWpCmd(subcommand []string) (string, error) {
+	if fpm {
+		return runWpFpmCmd(subcommand)
+	} else {
+		return runWpCliCmd(subcommand)
+	}
+}
+
+func runWpFpmCmd(subcommand []string) (string, error) {
+	
+	path := fpmUrl.Path
+	host := fpmUrl.Host
+	
+	if path == "" || fpmUrl.Scheme == "unix" {
+		path = fpmUrl.Fragment
+	}
+	
+	if fpmUrl.Scheme == "unix" {
+		host = fpmUrl.Path
+	}
+	
+	fcgi, err := fcgiclient.Dial(fpmUrl.Scheme, host)
+	if err != nil {
+		log.Printf("fastcgi Dial failed: %s\n", err)
+	}
+
+	resp, err := fcgi.PostForm(buildFpmEnv(), buildFpmQuery(subcommand))
+	if err != nil {
+		logger.Printf("Failed to POST to FPM: %s\n", err)
+	}
+
+	content, err = ioutil.ReadAll(resp.Body)
+	if err != nil {
+		logger.Printf("Could not read FPM response: %s\n", err)
+	}
+	logger.Printf("content: %s\n", string(content))
+}
+
+func buildFpmEnv() (map[string]string) {
+	env := make(map[string]string)
+	// Need to run the following in an init container:
+	// mkdir $WP_CLI_ROOT && \
+	// cp /usr/local/bin/wp /tmp/wp.phar && \
+	// phar extract -f /tmp/wp.phar $WP_CLI_ROOT
+	env["WP_CLI_ROOT"] = "/usr/local/wp-cli"
+	// This script needs to be injected into $WP_CLI_ROOT by wpvip-operator
+	env["SCRIPT_FILENAME"] = "/var/www/wp-content/mu-plugins/cron-control/includes/fpm-cron-runner.php"
+	env["GATEWAY_INTERFACE"] = "FastCGI/1.0"
+	env["REQUEST_METHOD"] = "POST"
+	env["CONTENT_TYPE"] = "application/x-www-form-urlencoded"
+	return env
+}
+
+func buildFpmQuery(subcommand []string) string {
+	queryData := url.Values{}
+	for _, s := range subcommand {
+		kv := strings.Split(s, "=")
+		k := kv[0]
+		v  := ""
+		if len(kv) == 1 && strings.HasPrefix(k, "--") {
+			queryData.Add(kv[0], "true")
+		} else if len(kv) == 1 {
+			queryData.Add("subcommands", kv[0])
+		} else if len(kv) == 2 {
+			queryData.Add(kv[0], kv[1])
+		} else if len(kv) > 2 {
```

Comment:
You can use `strings.SplitN` to avoid this case

Reply:
I don't see how that would do what this is doing. This is distinguishing between positional args, flags, and kwargs.